src/charm.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charmed Kubernetes Operator for the PostgreSQL database."""
+
+import logging
+import secrets
+import string
+
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresqlOperatorCharm(CharmBase):
+    """Charmed Operator for the PostgreSQL database."""
+
+    _stored = StoredState()
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self._postgresql_service = "postgresql"
+
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.postgresql_pebble_ready, self._on_postgresql_pebble_ready)
+
+    def _on_install(self, _):
+        """Event handler for InstallEvent."""
+        # TODO: change to peer/leader data bag when relations are implemented.
+        self._stored.postgres_password = self._new_password()
+
+    def _on_config_changed(self, _):
+        """Handle the config-changed event."""
+        # TODO: placeholder method to implement logic specific to configuration change.
+        pass
+
+    def _on_postgresql_pebble_ready(self, _) -> None:
+        """Event handler for on PebbleReadyEvent."""
+        # TODO: move this code to an "_update_layer" method in order to also utilize it in
+        # config-changed hook.
+        # Get the postgresql container so we can configure/manipulate it.
+        container = self.unit.get_container(self._postgresql_service)
+        # Create a new config layer.
+        layer = self._postgresql_layer()
+
+        if container.can_connect():
+            # Get the current config.
+            services = container.get_plan().services
+            # Check if there are any changes to services.
+            if services != layer.services:
```

Comment:
IMHO, we can improve readability and style with something like:

```python
#  These names make it more clear to understand what is what
new_layer = self._postgresql_layer().services
if container.can_connect():
    current_layer = container.get_plan().services

# Now the statement compares two of the same kind
if current_layer != new_layer:
    ...
```

Reply:
Thanks, Daniela! Yes, I agree that it gets more readable if we do that changes. I updated the code and changed something in your suggestion in order to make it clear for others that the variables represent layers/plan instead of services (and be able use `new_layer` in the `add_layer` call):

```python
new_layer = self._postgresql_layer()
if container.can_connect():
    current_layer = container.get_plan()
    if current_layer.services != new_layer.services:
        container.add_layer(self._postgresql_service, new_layer, combine=True)
```